crates/net/network/src/transactions.rs

```diff
@@ -209,6 +214,10 @@ where
         if self.network.is_initially_syncing() {
             return
         }
+        #[cfg(feature = "optimism")]
+        if self.network.disable_tx_gossip() {
+            return
+        }
 
         trace!(target: "net::tx", "Start propagating transactions");
 
```

Comment:
I think we may as well also add the short circuit for disabled tx gossip to the `propogate_transactions` method (below) in case its ever called.

Reply:
added